tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -600,6 +600,52 @@ bool containOnlyVecWithFourFloats(QualType type, bool use16Bit) {
   return false;
 }
 
+// Evaluates whether a given QualType is in fact an array and unroll accordingly.
+// Will return the the appropriate RecordType for the provided QualType. Will be 
+// the actual type of the array element, if it's really an array.
+template <size_t SmallVectorSize>
+QualType unrollMultiDimensionalArray(
+    const ASTContext &astContext, const QualType &startType,
+    llvm::SmallVector<int, SmallVectorSize> *arraySizes
+) {
```

Comment:
Can you please format this whole PR with clang-format? You can do that with something like `git diff main -U0 | clang-format-diff.py -i -p1`. See https://clang.llvm.org/docs/ClangFormat.html#script-for-patch-reformatting

Reply:
It was updated to python3 over a year ago, so it should be in llvm 13 and newer: https://github.com/llvm/llvm-project/blame/5179f885d159a01e00c208e3255052fe0fde06bf/clang/tools/clang-format/clang-format-diff.py#L1